### PR TITLE
Improve dotnet-test eval coverage and efficiency

### PR DESCRIPTION
**Pull request by Evangelink**
## Summary

Addresses remaining high-confidence work from #899:

- bounds code-testing research and implementation to the requested scope
- removes default discovery fan-out and repeated workspace inventories
- reuses research artifacts and loads deep examples only when needed
- strengthens canonical Vally `eval.yaml` coverage for code-testing-agent, find-untested-sources, and grade-tests
- covers TypeScript test generation plus the Python sparse-workspace integrity guardrail
- adds discriminating C# namespace and TypeScript source-pairing fixtures, including nested-namespace disambiguation
- removes fixture trees left orphaned by retired scenarios

The four remaining coverage gaps are non-user-invocable reference skills and need dependency-level evaluation design rather than direct-activation tests.

## Checks

- `skill-validator check --plugin ./plugins/dotnet-test` passes for all 20 skills and 10 agents
- eval YAML and fixture references parse successfully
- the expanded namespace fixture builds and Roslyn reports the expected paired/unpaired paths
- final Vally evaluation passes all three changed skills:
  - code-testing-agent: +40.0% [40.0%, 40.0%], 2/0/0
  - find-untested-sources: +40.0% [40.0%, 40.0%], 3/0/0
  - grade-tests: +70.0% [14.9%, 125.1%], 4/0/0
- the independent five-model cross-family run reported positive preference in every cell; code-testing-agent was credibly better for 3/5 models and find-untested-sources for 2/5

Refs #899

### CODE REVIEW

File: tests/dotnet-test/grade-tests/eval.vally.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    graders:
+      - type: output-matches
+        config:
+          pattern: \|\s*Test\s*\|\s*Grade\s*\|
+      - type: output-matches
+        config:
+          pattern: PlaceOrder_ValidItems_AssignsOrderIdAndPersistsOrder.*\|\s*A\s*\|
+      - type: output-matches
+        config:
+          pattern: CancelOrder_ExistingOrder_Works.*\|\s*F\s*\|
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/grade-tests/eval.vally.yaml]: Addressed: the Vally eval now explicitly requires `GetOrderById_ExistingId_ReturnsOrder` to appear with grade C, so omitting or misgrading that method fails the objective grader.

File: tests/dotnet-test/find-untested-sources/eval.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          source: "fixtures/pairing-repo/src/Store/OrderProcessor.cs"
+        - path: "PairingRepo/tests/Store.Tests/Store.Tests.csproj"
+          source: "fixtures/pairing-repo/tests/Store.Tests/Store.Tests.csproj"
+        - path: "PairingRepo/tests/Store.Tests/CustomerServiceTests.cs"
+          source: "fixtures/pairing-repo/tests/Store.Tests/CustomerServiceTests.cs"
+    assertions:
+      - type: "output_contains"
+        value: "OrderProcessor.cs"
+      - type: "output_contains"
+        value: "OrderProcessorTests.cs"
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/find-untested-sources/eval.yaml]: Addressed: the skill-validator assertion now matches the full separator-tolerant path `PairingRepo/tests/Store.Tests/OrderProcessorTests.cs` instead of only the filename.

File: tests/dotnet-test/find-untested-sources/eval.vally.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+          dest: PairingRepo/tests/Store.Tests/Store.Tests.csproj
+        - src: fixtures/pairing-repo/tests/Store.Tests/CustomerServiceTests.cs
+          dest: PairingRepo/tests/Store.Tests/CustomerServiceTests.cs
+    graders:
+      - type: output-matches
+        config:
+          pattern: OrderProcessor\.cs
+      - type: output-matches
+        config:
+          pattern: OrderProcessorTests\.cs
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/find-untested-sources/eval.vally.yaml]: Addressed: the Vally grader now matches the full separator-tolerant `PairingRepo/tests/Store.Tests/OrderProcessorTests.cs` path.

File: plugins/dotnet-test/agents/code-testing-researcher.agent.md [hunk 4]
```diff
@@ -54,17 +58,16 @@ Based on files found:
 - Did user ask for specific files, folders, methods, or entire project?
 - If specific scope is mentioned, focus research on that area. If not, analyze entire codebase.
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply plugins/dotnet-test/agents/code-testing-researcher.agent.md:64]: Addressed: when scope is omitted, research is now bounded to the nearest project/package manifest; it must not inventory sibling projects or expand to the entire workspace.

File: plugins/dotnet-test/skills/find-untested-sources/SKILL.md [hunk 5]
```diff
@@ -1,20 +1,13 @@
...
-  USE FOR: where to write tests next, which files have no tests, find untested
-  code, build a source-to-test pairing map, prioritized test-gap worklist.
-  DO NOT USE FOR: line/branch coverage or CRAP risk (use coverage-analysis);
-  whether existing tests are strong (use test-gap-analysis or assertion-quality).
+  MANDATORY for static requests to find, identify, or list untested source files
+  or modules, sources without tests, source-to-test pairing, test-gap worklists,
+  or suggested test locations. Invoke even for a tiny package; do not substitute
+  manual globbing. Uses Roslyn for C#/.NET and tree-sitter for Python, TS/JS, Go,
+  Java, Rust, and Ruby. DO NOT USE FOR: line/branch coverage, CRAP risk, or
+  grading existing tests.
```
Comment (AbhitejJohn) [inline plugins/dotnet-test/skills/find-untested-sources/SKILL.md:9]: Removing `disable-model-invocation: true` re-adds this skill to the model-facing menu. Recomputing the dotnet-test plugin's rendered skill-menu size the way the validator does (`SkillProfiler.RenderedSkillMenuCost`): main is 14,671/15,000, this PR is **14,943/15,000** — only 57 chars of headroom. It passes today, but the next description edit that adds >57 chars will trip the aggregate check, and at runtime the alphabetically-last skill (`writing-mstest-tests`) is the one that gets truncated. Since the PR notes the validator couldn't be rerun on the final state, please run a clean-worktree skill-validator check before merging.

File: plugins/dotnet-test/skills/code-testing-agent/SKILL.md [hunk 6]
```diff
@@ -94,12 +89,14 @@ The Test Generator will manage the entire pipeline automatically.
...
-3. For C# multi-file scopes, run `find-untested-sources` once and consume its `source_to_tests`, `untested`, and `suggested_test_path` output; do not repeat that pairing manually.
-4. Plan each target file once, then implement phases sequentially.
-5. Build and test the narrow target during fix cycles; run workspace-level validation once at the end.
-6. Read a language example from `code-testing-extensions` only when the repository has no representative tests and the base extension is insufficient.
+1. Turn every explicit user requirement into a checklist before implementation. Include requested layers, collaborators to mock, boundary cases, integrations, coverage thresholds, and report artifacts.
+2. Research only the requested module or project and write the checklist plus a compact target inventory to `.testagent/research.md`.
+3. Reuse manifests, symbol references, and deterministic pairing tools instead of reading every source and test file.
+4. For C# multi-file scopes, run `find-untested-sources` once and consume its `source_to_tests`, `untested`, and `suggested_test_path` output; do not repeat that pairing manually.
```
Comment (AbhitejJohn) [inline plugins/dotnet-test/skills/code-testing-agent/SKILL.md]: This still scopes `find-untested-sources` to "C# multi-file scopes", but the skill's description in this same PR now advertises it as MANDATORY with tree-sitter support for Python, TS/JS, Go, Java, Rust, and Ruby — and a new eval expects Python activation. `code-testing-researcher.agent.md` similarly says to "Fall back to bounded manual discovery … when the code is non-C#." Net effect: the pipeline will never use the polyglot engine this PR just enabled. Either broaden these caller instructions to all supported languages, or narrow the skill description back to C#.

File: tests/dotnet-test/find-untested-sources/eval.vally.yaml [hunk 7]
```diff
@@ -28,7 +28,7 @@ stimuli:
           pattern: OrderProcessor\.cs
       - type: output-matches
         config:
-          pattern: OrderProcessorTests\.cs
+          pattern: PairingRepo[\\/]tests[\\/]Store\.Tests[\\/]OrderProcessorTests\.cs
```
Comment (AbhitejJohn) [inline tests/dotnet-test/find-untested-sources/eval.vally.yaml]: The new "Identify unpaired Python source files" scenario was added to `eval.yaml` but not here — the Vally eval still covers only the C# scenario. Consider porting the Python scenario for parity across both eval formats.

**Review by AbhitejJohn — COMMENTED** [review]
Multi-model review (GPT-5.5 + Gemini 3.1 Pro + manual verification). Substance looks good; two items are worth resolving before merge. Details inline.

- **Skill-menu budget**: this PR takes dotnet-test from 14,671 to 14,943 of the 15,000 rendered-menu cap (57 chars headroom). Please rerun the skill-validator on a clean worktree since the PR notes it couldn't be verified on the final state.
- **Polyglot vs. C#-only callers**: the re-enabled polyglot description isn't matched by the orchestrator instructions, which still restrict `find-untested-sources` to C#.

Verified clean: regex escaping in both eval formats, Python `suggested_test_path` (`test_pricing.py`), no dangling refs from the removed Examples section, generator `user-invocable: false` consistent with README, and `additional_required_skills` usage in grade-tests.

**Comment by AbhitejJohn** [comment]
I merged latest `main` and resolved the conflicts, which came from main's completed Vally-harness migration (single `eval.yaml` per skill, zero `eval.vally.yaml`). Net result: `mergeable` is now `MERGEABLE` and every `eval.vally.yaml` is gone repo-wide.

How I reconciled the three conflicting dotnet-test evals (kept this PR's intent + main's invariant):

- **code-testing-agent** — clean auto-merge: main's canonical `eval.yaml` + this PR's four "standard test-generation workflow" prompt additions.
- **find-untested-sources** — adopted main's canonical `eval.yaml` and folded in this PR's tightened `OrderProcessorTests` path grader. Deleted the `eval.vally.yaml`.
- **grade-tests** — this needs your eyes. main deliberately **redesigned** grade-tests down to 2 scenarios (MSTest curated + no-list) and dropped the `additional_required_skills`/`pairwise` mechanisms this PR relied on. This PR instead enriches the old-schema eval to 5 scenarios. I kept main's redesigned MSTest + no-list scenarios and **re-authored your three polyglot coverage scenarios (pytest, Go table-driven, production-unavailable) in the canonical Vally schema** so the coverage isn't lost.

Two judgment calls I'd like you to confirm:
1. **grade-tests MSTest scenario**: I used main's trimmed 4-method version (with main's shorter rubric) rather than your 7-method version (which added `PlaceOrder_EmptyItems`, `SerializeOrderId_RoundTrip`, `PlaceOrder_LongRunning`). If you want the richer MSTest coverage back, I can re-express those in the canonical schema too.
2. **find-untested-sources Python scenario**: your old-schema `eval.yaml` flipped scenario 2 to expect Python pairing support, but neither main's `eval.yaml` nor this branch's `eval.vally.yaml` carried a Python scenario, so I dropped it. Happy to re-add it in canonical form if Python pairing is intended to be covered.

All re-authored YAML parses and matches main's canonical grader vocabulary; `markdownlint` is clean on the changed docs. I couldn't run the skill-validator `check` locally (it pins SDK `11.0.100-preview.3`, not installed here) — CI will cover that.

File: tests/dotnet-test/find-untested-sources/eval.yaml [hunk 8]
```diff
@@ -32,23 +32,25 @@ scenarios:
...
                 return price * quantity
     assertions:
-      - type: "output_matches"
-        pattern: "(C#|Python|not supported|C#-only|polyglot)"
+      - type: "output_contains"
+        value: "pricing.py"
+      - type: "output_contains"
+        value: "test_pricing.py"
       - type: "exit_success"
     rubric:
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/find-untested-sources/eval.yaml]: Good catch — fixed in 469855593. The Python scenario now rejects standalone pytest execution commands and coverage commands.

File: plugins/dotnet-test/skills/code-testing-agent/SKILL.md (hunk 6 above)
Comment (Evangelink) [reply plugins/dotnet-test/skills/code-testing-agent/SKILL.md]: Fixed in 469855593. Both the orchestrator skill and researcher agent now invoke `find-untested-sources` for every supported language and only fall back for unavailable or unsupported cases.

File: plugins/dotnet-test/skills/find-untested-sources/SKILL.md (hunk 5 above)
Comment (Evangelink) [reply plugins/dotnet-test/skills/find-untested-sources/SKILL.md:9]: Verified on clean commit 469855593 with the published `skill-validator check --plugin ./plugins/dotnet-test`; all 20 skills and 10 agents pass.

File: tests/dotnet-test/find-untested-sources/eval.vally.yaml (hunk 7 above)
Comment (Evangelink) [reply tests/dotnet-test/find-untested-sources/eval.vally.yaml]: Fixed in 469855593. Since the Vally migration consolidated specs into the sole `eval.yaml`, I added the Python scenario and fixture to that canonical file.

File: tests/dotnet-test/grade-tests/eval.yaml [hunk 9]
```diff
@@ -74,3 +74,154 @@ stimuli:
...
+          pattern: (test_calculate_total_with_discount_applies_discount_and_returns_rounded_amount.*\|\s*A\s*\|)
+      - type: output-matches
+        config:
+          pattern: (test_calculate_total_with_negative_discount_raises_value_error.*\|\s*A\s*\|)
+      - type: output-matches
+        config:
+          pattern: (test_clear_cart_works.*\|\s*F\s*\|)
+      - type: output-matches
+        config:
+          pattern: (test_async_checkout_completes.*\|\s*F\s*\|)
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/grade-tests/eval.yaml:71]: Good catch — fixed in 469855593. The pytest scenario now objectively requires `test_get_cart_returns_cart` to receive grade C.

File: tests/dotnet-test/grade-tests/eval.yaml [hunk 10]
```diff
@@ -67,9 +101,155 @@ stimuli:
...
+        config:
+          pattern: (Charge_NegativeAmount_ThrowsArgumentOutOfRange.*\|\s*A\s*\|)
+      - type: output-matches
+        config:
+          pattern: (Refund_ExistingCharge_ReturnsReceipt.*\|\s*C\s*\|)
+      - type: output-matches
+        config:
+          pattern: (Settle_PendingBatch_Runs.*\|\s*F\s*\|)
+      - type: exit-success
+      - type: prompt
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/grade-tests/eval.yaml:179]: Good catch — fixed in 5fc8eaa47. The scenario now objectively requires the exact `Production-dependent behavior coverage: Unverified` disclaimer.

File: tests/dotnet-test/find-untested-sources/eval.yaml [hunk 11]
```diff
@@ -28,7 +28,7 @@ stimuli:
           pattern: OrderProcessor\.cs
       - type: output-matches
         config:
-          pattern: OrderProcessorTests\.cs
+          pattern: PairingRepo[\\/]tests[\\/]Store\.Tests[\\/]OrderProcessorTests\.cs
       - type: output-not-matches
         config:
           pattern: (dotnet test|coverage\.cobertura|CRAP)
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/find-untested-sources/eval.yaml]: Fixed in 2822ca931. Both C# static-pairing scenarios now reject command-like `dotnet build` and `dotnet test` output.

File: tests/dotnet-test/find-untested-sources/eval.yaml [hunk 12]
```diff
@@ -37,5 +37,128 @@ stimuli:
...
+        config:
+          pattern: tax\.ts
+      - type: output-matches
+        config:
+          pattern: tax\.(test|spec)\.ts
+      - type: output-not-matches
+        config:
+          pattern: (?i)pricing\.ts.{0,100}(untested|unpaired|no test)
+      - type: exit-success
+      - type: prompt
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/find-untested-sources/eval.yaml:84]: Fixed in 2822ca931. The TypeScript scenario now rejects npm/pnpm/yarn install and test commands, and the Go scenario rejects `go test` commands.

File: plugins/dotnet-test/skills/find-untested-sources/SKILL.md [hunk 13]
```diff
@@ -251,13 +244,19 @@ orders-of-magnitude lower cost than coverage. Known gaps:
 For these cases, run actual coverage (`coverage-analysis`) on the unpaired
 candidates the agent has already triaged.
 
+Always label the final result as a static pairing heuristic, not evidence of
+line or branch coverage. Include that caveat even when every requested source
+file has an obvious matching or missing test.
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply plugins/dotnet-test/skills/find-untested-sources/SKILL.md:266]: Fixed in 2822ca931. The C# scenario now objectively requires both a static pairing/heuristic label and the line/branch coverage limitation; this also separates the prior skilled outputs from baseline.

File: tests/dotnet-test/code-testing-agent/eval.yaml [hunk 14]
```diff
@@ -209,6 +205,37 @@ stimuli:
...
+        config:
+          command: sh -c "cd fixtures/python-flask-tasks && python3 -m pip install -e '.[test]' --quiet && python3 -m pytest -q -o addopts='' tests"
+          expected_exit_code: 0
+          timeout: 10m
+      - type: run-command
+        config:
+          command: sh -c "find fixtures/python-flask-tasks/tests -type f -name 'test_*.py' -print -quit | grep -q test_"
+          expected_exit_code: 0
+          timeout: 1m
+      - type: prompt
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/code-testing-agent/eval.yaml]: Addressed in 007584b19: the redundant focused TaskService scenario was removed after repeated runs showed it did not differentiate the skill, so the unenforced production-file constraint no longer exists.

File: tests/dotnet-test/grade-tests/eval.yaml [hunk 15]
```diff
@@ -40,6 +41,8 @@ stimuli:
       - type: prompt
     rubric:
       - Produced one compact table row for every requested test with grade, score band, and a specific note
+      - Used the stable `Test | Grade | Band | Notes` table schema with letter grades and score bands rather than a
+        different column layout or point scores
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/grade-tests/eval.yaml]: Good catch — fixed in 2d1442411. Every remaining per-test grading scenario now objectively requires the complete `Test | Grade | Band | Notes` header.

File: tests/dotnet-test/find-untested-sources/eval.yaml [hunk 16]
```diff
@@ -4,38 +4,80 @@ type: capability
...
-          pattern: OrderProcessor\.cs
+          pattern: Beta[\\/]Settings\.cs
       - type: output-matches
         config:
-          pattern: OrderProcessorTests\.cs
+          pattern: (?:NamespaceCollision[\\/])?tests[\\/]NamespaceCollision\.Tests[\\/]Beta[\\/]SettingsTests\.cs
       - type: output-not-matches
         config:
-          pattern: (dotnet test|coverage\.cobertura|CRAP)
+          pattern: (?i)Alpha[\\/]Settings\.cs.{0,100}(untested|unpaired|no test)
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/find-untested-sources/eval.yaml]: Fixed in 2d1442411. The namespace-disambiguation scenario now requires `AlphaSettingsTests.cs` in addition to rejecting Alpha as unpaired.

File: tests/dotnet-test/find-untested-sources/eval.yaml [hunk 17]
```diff
@@ -4,38 +4,80 @@ type: capability
...
+    graders:
+      - type: output-matches
+        config:
+          pattern: tax\.ts
+      - type: output-matches
+        config:
+          pattern: tax\.(test|spec)\.ts
+      - type: output-not-matches
+        config:
+          pattern: (?i)pricing\.ts.{0,100}(untested|unpaired|no test)
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/find-untested-sources/eval.yaml:73]: Fixed in 2d1442411. The TypeScript scenario now objectively requires the paired `pricing.test.ts` evidence.

File: tests/dotnet-test/grade-tests/eval.yaml [hunk 18]
```diff
@@ -8,7 +8,8 @@ stimuli:
     prompt: |
       Grade these test methods individually and return a compact markdown table
       with one row per test, a letter grade, score band, and one-line note. Do
-      not modify files.
+      not modify files. Apply this repository's established per-test grading
+      policy rather than inventing a new scoring scale.
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/grade-tests/eval.yaml]: Updated the PR description to describe the canonical Vally `eval.yaml` format used after the repository-wide migration; no separate `eval.vally.yaml` is expected.

File: tests/dotnet-test/find-untested-sources/eval.yaml [hunk 19]
```diff
@@ -4,38 +4,80 @@ type: capability
...
-  - name: Identify unpaired C# source files and suggest a test location
+  - name: Disambiguate duplicate C# type names by namespace
     prompt: |
-      Find the C# source files under PairingRepo that have no test file
-      referencing their declared types. Prioritize the largest untested file and
-      tell me where its test file should go. I only need the static pairing
-      result; do not build the projects or collect coverage.
+      Find the C# source file under NamespaceCollision that is not referenced by
+      any test and suggest the exact test path in the existing test project. Do
+      not build or collect coverage.
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/find-untested-sources/eval.yaml:11]: Updated the PR description to describe the canonical Vally `eval.yaml` format used after the repository-wide migration; no separate `eval.vally.yaml` is expected.

**Comment by AbhitejJohn** [comment]
@Evangelink Thanks for this — and for the sustained iteration on the review threads; a lot of the grader-tightness feedback (full-path assertions, pytest/npm/`dotnet build` execution guards, the objective grade-C and disclaimer assertions) has already landed, and it shows. I ran the change through several independent AI code reviews plus an automated skill-quality check, and a cross-family evaluation (five models doing the tasks, scored by a separate model). Sharing what came back.

**Cross-family evaluation.** Every one of the fifteen cells (three skills × five models) came back with a *positive* preference for the new version — no model preferred the baseline anywhere; the non-passes are conclusiveness issues (small samples with CIs crossing 0, or unmatched trajectories), not regressions. Full run: https://github.com/dotnet/skills/actions/runs/30051741705 *(dispatched on the PR head at the time, `007584b1`; the suite has had further grader fixes since, so treat these as directional.)*

`grade-tests` — the strongest skill (4/5 credibly better):

| Executor model | Judge | Result | Win rate (W/T/L) | Mean preference [95% CI] |
|---|---|---|---|---|
| Opus | GPT | ✅ credibly better | 91.7% (11/1/0) | +86.7% [+66.2, +107.2] |
| GPT | Opus | ✅ credibly better | 100% (12/0/0) | +100.0% [+100, +100] |
| Sonnet 4.6 | Opus | positive, inconclusive (unmatched) | 100% (11/0/0) | +94.5% [+82.4, +106.7] |
| Haiku | Opus | ✅ credibly better | 91.7% (11/0/1) | +83.3% [+56.3, +110.3] |
| MAI Flash | Opus | ✅ credibly better | 75.0% (9/0/3) | +55.0% [+2.4, +107.6] |

`code-testing-agent` — solid (3/5 credibly better; Opus/MAI positive but not statistically credible):

| Executor model | Judge | Result | Win rate (W/T/L) | Mean preference [95% CI] |
|---|---|---|---|---|
| Opus | GPT | not credible (CI includes 0) | 50.0% (3/2/1) | +13.3% [−20.9, +47.6] |
| GPT | Opus | ✅ credibly better | 100% (6/0/0) | +50.0% [+24.3, +75.7] |
| Sonnet 4.6 | Opus | ✅ credibly better | 83.3% (5/1/0) | +33.3% [+16.2, +50.5] |
| Haiku | Opus | ✅ credibly better | 100% (6/0/0) | +70.0% [+35.5, +104.5] |
| MAI Flash | Opus | not credible (CI includes 0) | 66.7% (4/2/0) | +46.7% [−0.6, +93.9] |

`find-untested-sources` — the weakest (2/5 credible; the rest positive but CIs cross 0):

| Executor model | Judge | Result | Win rate (W/T/L) | Mean preference [95% CI] |
|---|---|---|---|---|
| Opus | GPT | not credible (CI includes 0) | 83.3% (5/0/1) | +26.7% [−7.6, +60.9] |
| GPT | Opus | ✅ credibly better | 83.3% (5/1/0) | +43.3% [+9.7, +77.0] |
| Sonnet 4.6 | Opus | ✅ credibly better | 83.3% (5/1/0) | +33.3% [+16.2, +50.5] |
| Haiku | Opus | not credible (CI includes 0) | 33.3% (2/3/1) | +6.7% [−24.9, +38.3] |
| MAI Flash | Opus | not credible (CI includes 0) | 50.0% (3/3/0) | +20.0% [−3.0, +43.0] |

**Findings, grouped by severity.**

**🔴 Blocking**
- **`code-testing-agent`'s capability eval is now a single TypeScript scenario.** At the current head, `tests/dotnet-test/code-testing-agent/eval.yaml` contains only "Generate Vitest tests for the shopping-cart library." The C# generation scenario now runs only through the skill-validator harness (not this capability eval), and two things dropped out entirely: the **Python pytest-generation** scenario and the **workspace-integrity guardrail** ("does not revert a gutted-looking workspace"). That last one is the one I'd really restore — this same PR *strengthens* the skill's gutted-workspace behavior, so deleting the scenario that checks it means the behavior you're hardening is no longer tested. For a PR aimed at improving coverage, bringing back at least the integrity guardrail (and ideally one C#- or Python-generation cell) would keep the flagship skill honest.

**🟡 Non-blocking**
- **Three orphan fixtures** are left behind by removed scenarios — referenced by zero scenarios at head: `code-testing-agent/.../python-flask-tasks`, `find-untested-sources/fixtures/python-package`, and `grade-tests/fixtures/mixed-quality`. Either wire each back into a scenario or delete it so the suite stays clean and the intent is unambiguous.
- **`find-untested-sources` is the soft spot** in the evaluation and it runs only ~6 trials per cell versus `grade-tests`' ~12. The direction is positive everywhere but only two models clear the bar (Haiku is nearly flat at +6.7%). More trials — and/or a bit more guidance for the weaker cells — would firm up the signal.

**✅ What's clearly good**
- Removing the `// Expected grade: A`-style banner comments from the grading fixtures closes a real answer-leakage hole — nice fix.
- The new `namespace-collision` (C#) and `typescript-pairing` scenarios genuinely strengthen `find-untested-sources`.
- Net across all three skills the new version is preferred everywhere, and `grade-tests` is a standout.

Net: the direction and most of the execution are strong, and the earlier grader concerns are largely handled. The one I'd resolve before merge is the `code-testing-agent` coverage collapse (especially the integrity guardrail); the orphan-fixture cleanup and a few more `find-untested-sources` trials are easy follow-ups. Thanks again for the thorough work here!

**Comment by Evangelink** [comment]
@AbhitejJohn Addressed in db1e3e406. Restored the workspace-integrity pytest generation guardrail, added a third `find-untested-sources` cell that exercises Roslyn nested-namespace disambiguation, and removed the three orphan fixture trees (`python-flask-tasks`, `python-package`, and `mixed-quality`). The expanded namespace fixture builds and both Roslyn pairing cases return the expected paired/unpaired paths; the full dotnet-test skill-validator check passes.

File: tests/dotnet-test/code-testing-agent/eval.yaml [hunk 20]
```diff
@@ -4,125 +4,6 @@ type: capability
...
-        sqlite3.connect(':memory:') fixture), covering at least add/get, update of tags, and delete
-      - Tests pin down behavior with concrete expected values — minimal use of `assert x is not None`-only assertions;
-        tests assert on actual returned `Task` fields (id, title, status, due_at, priority) rather than
-        truthy/existence-only checks
-      - At least one TaskService validation test exercises a *combination* of properties (e.g., over-length title AND
-        invalid priority, or tag normalization on a Task with naive due_at) rather than only single-property tests,
-        exercising property intersections
-
   - name: Generate Vitest tests for the shopping-cart library (TypeScript polyglot)
     prompt: |
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/code-testing-agent/eval.yaml:8]: Good catch — removed the unreferenced ContosoUniversity fixture in commit 020d19640. No references remain under the dotnet-test tests or plugin.

[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/code-testing-agent/eval.yaml:8]: Correction: the cleanup commit is f5e892e46 (the prior reply contained a stale short SHA). The ContosoUniversity fixture is removed and no references remain.

File: tests/dotnet-test/grade-tests/eval.yaml [hunk 21]
```diff
@@ -4,72 +4,181 @@ type: capability
...
+  - name: Grade pytest test methods using the same rubric
     prompt: |
-      Grade these test methods individually and return a compact markdown table
-      with one row per test, a letter grade, score band, and one-line note. Do
-      not modify files.
+      Please grade each of the following pytest test functions
+      individually for test quality and produce a compact per-test
+      table (one row per test) plus a short summary. They live in
+      `tests/test_shopping_cart.py`. Apply this repository's established
+      per-test grading policy and do not modify any files.
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/grade-tests/eval.yaml]: Good catch — fixed in 70a8f545b. The pytest scenario now explicitly limits grading to observable test-body issues and objectively requires the standard `Production-dependent behavior coverage: Unverified` disclaimer.

[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/grade-tests/eval.yaml]: Correction: the fix commit is f5ac754b1 (the prior reply contained a stale short SHA). The pytest context and objective Unverified assertion are present there.

**Review by JanKrivanek — APPROVED** [review]
